### PR TITLE
Loot Table Rework

### PR DESCRIPTION
**Pull request by Nikolassparrow**
https://www.notion.so/H1Z1-Dec-2016-loot-tables-bf68dc58bc634c21b46be99e20fdaa53

I made an entire documentation about the loot tables from the December 22nd, 2016 update with the help from Kentin. Essentially, this PR is a rework of the current loot tables as there are many items missing, didn't spawn, lack of spawns etc. This is NOT meant to be a one-all-fix-all for the loot system as the architecture for it is flawed. All the changes that were made in this came from YouTube videos from that time period, and were also fact-checked with players around the H1EMU community, although their opinions can sometimes be subjective and biased, it was a good foundation to get started with. 

There are also some other QOL things that were added that I had noticed while researching this.

These are all the changes this PR makes:

- Reworks the loot tables to contain items that were present in the December 22nd, 2016 update. 
- Add other tool cabinet props that spawn in car shops.
- Update container search times to match their 2016 counter-parts
- Change hotwire to match the 7 seconds it was in 2016 (https://www.youtube.com/watch?v=CxMj5F_rpwM - 0:15)
- Added buffed weapon lockers (more info in the documentation)
- Added createContainerLoot to the respawnLoot command (from an old branch that never got merged - thank you jason)
- Changed some cabinets around to match their 2016 counter-parts (more info in the documentation - File cabinets section)
- Change canned foods eat time to 4 seconds to match the 2016 time. (https://www.youtube.com/watch?v=mUm95wfBj94 - 1:39:37)
- Changed resting time to 20 seconds from 30 seconds (https://www.youtube.com/watch?v=aFpSNGow3QI - 1:17:15)
- Added some missing buffed crate spawns (blue containers south map and some isolated camps around the map)
- Added search all cabinets interaction string for cabinets + fridge type
- Fixed Ellohime and Neko-Chan conveys not appearing on the characters feet when equipped.
- Added worn handcuffs, but they aren't functional (in my opinion, it probably shouldn't be added in the game 😂😂)

Please test this PR thoroughly before pushing to live servers, and let me know your feedback!

### CODE REVIEW

File: data/2016/dataSources/ServerItemDefinitions.json [hunk 1]
```diff
@@ -5599,7 +5599,7 @@
         "PASSIVE_ABILITY_ID": 0,
         "COST": 0,
         "ITEM_CLASS": 16053,
-        "MAX_STACK_SIZE": 9999,
```
Comment (avcio9) [inline data/2016/dataSources/ServerItemDefinitions.json:5602]: Thats just gonna lower the QOL, even if it worked like that back in the day

**Review by avcio9 — CHANGES_REQUESTED** [review]
We are supposed to make a survival game so why turning up all spawns, specially weapons when there is no issue in finding them, if anything the chances should be lowered + the weapons you added to residential spawners etc removed.

There was no bug in container create loot, you got confused cause we use the same type for world item and container items spawners. 
Your change made most of lootable props have 100% chance of refreshing and spawning all possible items. For example every wrecked vehicle spawns scrap, sheet and gun repair kit. It has to be reverted and please test every change you make.

File: data/2016/dataSources/ServerItemDefinitions.json (hunk 1 above)
Comment (Nikolassparrow) [reply data/2016/dataSources/ServerItemDefinitions.json:5602]: Maybe having a max of 1 isn't great for QOL, but people are literally carrying around 10+ med kits, it makes bandages and other healing types completely useless. I don't think 1 is great either, maybe 3 or 4 max stack would be solid? People running around with 10+ is ridiculous though.

**Comment by Nikolassparrow** [comment]
> We are supposed to make a survival game so why turning up all spawns, specially weapons when there is no issue in finding them, if anything the chances should be lowered + the weapons you added to residential spawners etc removed.
> 
> There was no bug in container create loot, you got confused cause we use the same type for world item and container items spawners. Your change made most of lootable props have 100% chance of refreshing and spawning all possible items. For example every wrecked vehicle spawns scrap, sheet and gun repair kit. It has to be reverted and please test every change you make.
> 
> Other changes look good

Weapon spawns on H1EMU make 0 sense. I VIVIDLY remember in 2016 that if I wanted a good weapon, to go to either top of a hotel, romeros desk, shotguns under clerks desk, against the wall in Cleggs etc. Being able to find a gun in a residential building was SIGNIFICANTLY more rare to find then to the other places I just listed. Also some spawns just flat out didn't exist like Armoires/Dressers/WashingMachines/Dryers didn't spawn 308s or shotguns but on H1EMU they did. The reason I upped some of the spawns was to try and match the fact that gun spawns were more common in designated "hotspots".

**Comment by Nikolassparrow** [comment]
Also how was #1907 not a bug..? I tested all cabinet types several times before making that change and they would ALWAYS spawn with one or nothing inside of them. File cabinets were especially bad at this and caused them to essentially be a useless prop.

Another question too, why is the logic for creating container loot, regular loot, rewards for quests, NPC reward loot etc. all different logic..? Shouldn't their be some consistency with how loot tables work because as of right now, there's like 4 or 5 different ways H1EMU respawns loot.

**Comment by Nikolassparrow** [comment]
Avcio's ego is bruised and quote: "stop wasting my time." So need someone else to review this PR please 👍

File: data/2016/dataSources/ServerItemDefinitions.json (hunk 1 above)
Comment (avcio9) [reply data/2016/dataSources/ServerItemDefinitions.json:5602]: Make it more rare maybe

Comment (Z1Meme) [reply data/2016/dataSources/ServerItemDefinitions.json:5602]: Changing stack size doesnt change how many a player can carry, for that you'd have to change the bulk per item. Having a higher stack size is just a QOL change for people who want to carry around more medkits.

Comment (Nikolassparrow) [reply data/2016/dataSources/ServerItemDefinitions.json:5602]: I see. I'll revert this change and yea it's kinda pointless b/c even with 1 stack size ppl can just carry more. Changing the bulk would be better (if we want to change it in the future :D)

**Comment by QuentinGruber** [comment]
Hey @Nikolassparrow so my take on this:

You would need to create new PRS about this, especially splitting your change on loottable vs changes on enums & timer stuff. Why? Because it's easier to work with and also this PR is kinda dead due to the fact that it has been unsynced with the /dev branch for too long and #1907 isn't a thing on /dev anymore so testing spawn on this PR doesn't reflect how the changes would operate on the current codebase.

**Pull request closed without merging**